src/msw/listctrl.cpp

```diff
@@ -3275,20 +3280,64 @@ void wxListCtrl::OnPaint(wxPaintEvent& event)
         return;
     }
 
+    int visibleEnd = countPerPage;
+    visibleEnd -= 1;
+
     const long top = GetTopItem();
     const long bottom = wxMin(top + countPerPage, itemCount - 1);
 
+    wxRect clipRect;
+    dc.GetClippingBox(clipRect);
+
+    if (m_listRulesAlternateColourOnBlank && visibleEnd > bottom ) {
+        wxPaintDC dc2(this);
```

Comment:
We should reuse the existing `wxPaintDC` instead of creating another one, why is this necessary?

Reply:
It seems it would be simple to use `wxDC{Brush|Pen}Changer` with `dc` instead of creating another `wxPaintDC`.